hack/verify-description.sh

```diff
@@ -45,25 +45,38 @@ find_files() {
         -o -wholename './target' \
         -o -wholename '*/third_party/*' \
         -o -wholename '*/vendor/*' \
+        -o -wholename './pkg/*' \
       \) -prune \
     \) \
-    \( -wholename '*pkg/apis/*/v*/types.go' \
-       -o -wholename '*pkg/api/unversioned/types.go' \
-    \)
+    -wholename './staging/src/k8s.io/api/*/v*/types.go'
 }
 
 if [[ $# -eq 0 ]]; then
-  versioned_api_files=$(find_files | grep -E "pkg/.[^/]*/((v.[^/]*)|unversioned)/types\.go") || true
+  versioned_api_files=$(find_files) || true
 else
   versioned_api_files="${*}"
 fi
 
+# find_files had incorrect regexes which led to genswaggertypedocs never being invoked.
+# This led to many types.go have missing descriptions.
+# These types.go files are listed in hack/.descriptions_failures
+# Check that the file is in alphabetical order
+failure_file="${KUBE_ROOT}/hack/.descriptions_failures"
+kube::util::check-file-in-alphabetical-order "${failure_file}"
+
+failing_files=()
+while IFS='' read -r line; do failing_files+=("$line"); done < <(cat "$failure_file")
+
 for file in $versioned_api_files; do
-  $genswaggertypedocs -v -s "${file}" -f - || gen_swagger_result=$?
-  if [[ "${gen_swagger_result}" -ne "0" ]]; then
-    echo "API file: ${file} is missing: ${gen_swagger_result} descriptions"
-    result=1
+  if ! kube::util::array_contains "$file" "${failing_files[@]}"; then
+    gen_swagger_result=0
+    $genswaggertypedocs -v -s "${file}" -f - || gen_swagger_result=$?
+    if [[ "${gen_swagger_result}" -ne "0" ]]; then
```

Comment:
Drop quotes for numeric (as clearly indicated by `-ne` value 0.

`if [[ "${gen_swagger_result}" -ne 0 ]]; then`

Reply:
Thanks, done!